docs/application/native/guides/machine-learning/machine-learning-training.md

```diff
@@ -0,0 +1,469 @@
+
+status = ml_train_model_construct_with_conf(model_path, &model);
+if(status != ML_ERROR_NONE) {
+  // handle error
+}
+```
+
+### Construct a model on code
+
+An empty model can be constructed with `ml_train_model_construct`.
+
+## Configure the model
+
+After constructing a model, the model can be configured.
+
+> [!NOTE]
+> Example code written here reproduces the model description from [Create Model from INI Formatted File](#create-model-from-ini-formatted-file) except that followings are different:
+> * Relative path is changed to dynamic app resource and data path.
+> * Model related properties (that can only be set at compile or run phase)
+> * Demonstration about `ml_train_dataset_create_with_generator`, which cannot be covered in the description file.
+
+First, an empty model needs to be created:
+
+```c
+ml_train_model_h model;
+ml_train_model_construct(&model);
+```
+
+### Add a layer
+
+`ml_train_model_add_layer` appends a layer to the end of the graph in the model:
+
+```c
+int status = ML_ERROR_NONE;
+ml_train_layer_h layers[2];
+
+// create and add input layer
+status = ml_train_layer_create(&layers[0], ML_TRAIN_LAYER_TYPE_INPUT);
+if(status != ML_ERROR_NONE) {
+  // handle error
+}
+
+status = ml_train_layer_set_property(layers[0], "name=inputlayer",
+                                                "input_shape=1:1:62720",
+                                                "normalization=true", NULL);
+if(status != ML_ERROR_NONE) {
+  //handle error
+}
+status = ml_train_model_add_layer(model, layers[0]);
+
+// create and add fully connected layer
+status = ml_train_layer_create(&layers[1], ML_TRAIN_LAYER_TYPE_FC);
+status = ml_train_layer_set_property(layers[1], "name=outputlayer",
+                                                "unit=2",
+                                                "bias_initializer=zeros",
+                                                "weight_initializer=xavier_uniform",
+                                                "activation=sigmoid", NULL);
+status = ml_train_model_add_layer(model, layers[1]);
+```
+
+### Set an optimizer
+
+Creating and setting `optimizer` to a model can be done in the same manner as `layer`:
+
+```c
+status = ml_train_optimizer_create(&optimizer, ML_TRAIN_OPTIMIZER_TYPE_ADAM);
+status = ml_train_optimizer_set_property(optimizer, "learning_rate=0.0001",
+                                                    "decay_rate=0.96",
+                                                    "decay_steps=1000",
+                                                    "beta1=0.002",
+                                                    "beta2=0.001",
+                                                    "epsilon=1e-7", NULL);
+status = ml_train_model_set_optimizer(model, optimizer);
+```
+
+### Set a dataset
+
+There are two ways to create a dataset. One is from a file, and another one is from a callback.
+In either fashion, you need to provide streams of tensor data and one-hot-encoded labels.
```

Comment:
Suggestion. How about adding a simple description for "one-hot-encoded labels"?

Reply:
Added as below:

`One-hot-encoded label is a group of values with only one value representing the label,usually one bit high(1), and rests representing others, normally one bit low(0).`

Reply:
Thanks for adding the sentence.

@surbhig12 Would you please also review this additional sentence?

Beside, a space is needed before "usually"

Reply:
I think we need @zhoonit's check to the suggestion, as I'm not sure if there can be other value not high(1) or low(0). Wasn't only two bits (high(1) and low(0)) allowed, or used for the One-hot-encoded label?

Reply:
@mijin85cho 
It usually is 1 and 0. Each representing values can be other values though. https://www.tensorflow.org/api_docs/python/tf/one_hot.

@surbhig12. One hot encoded label is a label represented by a group of values, not other way around.

I deleted the definition as it seems hard to catch the meaning and it is a commonly known format for this domain, instead elaborated more on the previous sentence.

PTAL.

`
There are two ways to create a dataset. One is from a file, and another one is from a callback.
In either fashion, you need to provide streams of tensor data and arrays of values representing the label, usually one-hot-encoded.
`